packages/@okta/vuepress-site/docs/guides/terraform-create-custom-auth-server/main/index.md

```diff
@@ -0,0 +1,407 @@
+
+```hcl
+# For users in group writable_permissions_api, grant ALL scopes
+resource "okta_auth_server_policy_rule"
+    "rule_for_staff_with_writable_permissions" {
+  auth_server_id       = okta_auth_server.api_server.id
+  policy_id            = okta_auth_server_policy.example.id
+  status               = "ACTIVE"
+  name                 = "rule_for_staff_with_writable_permissions"
+  priority             = 1
+  group_whitelist      = [ okta_group.writable_permissions_api.id ]
+  grant_type_whitelist = ["authorization_code"]
+  scope_whitelist      = ["*"]
+}
+
+# For users in group readonly_permissions_api, grant ONLY the readapi scope
+resource "okta_auth_server_policy_rule"
+    "rule_for_staff_with_readonly_permissions" {
+  auth_server_id = okta_auth_server.api_server.id
+  policy_id            = okta_auth_server_policy.example.id
+  status               = "ACTIVE"
+  name                 = "rule_for_staff_with_readonly_permissions"
+  priority             = 2
+  group_whitelist      = [ okta_group.readonly_permissions_api.id ]
+  grant_type_whitelist = ["authorization_code"]
+  scope_whitelist      = [ okta_auth_server_scope.read_obj.name , "openid"]
+  depends_on = [
+    okta_auth_server_policy_rule.rule_for_staff_with_writable_permissions ]
+}
+```
+
+Because there’s more than one rule, all rules with a `priority` greater than `1` must declare explicit creation order with the `depends_on` meta attribute. See [link to the new syntax article, ideally the heading]
+
+### Help an external OAuth app validate the token issuer
+
+For a production system, it is a best practice for an external app that receives the token to confirm that the right issuer issued the token.
+
+There are multiple ways that external apps can verify that Okta minted tokens:
+
+* Use the Okta REST API to call the [authorization server introspect operation](/docs/reference/api/oidc/#introspect). The request takes an access token, ID token, refresh token, or device secret and returns a boolean that indicates whether the token is active.
+
+* Get the public key from the Okta-published JWKS endpoint. The key ID is in the header of the JWTs that Okta issues.
+
+### Test your external app with your authorization server
+
+Test your external app with your new authorization server. For initial testing, you can use Okta sample code to implement OAuth. See the [sample page](/docs/guides/sampleapp-oie-redirectauth/go/main/) and choose the platform from the selector in the upper right.
+
+The Okta configuration has no extra steps to support the OAuth Authorization Code flow with [Proof Key for Code Exchange](https://oauth.net/2/pkce/) (PKCE). Okta handles PKCE automatically if the OAuth request includes the challenge. However, you must properly configure your external OAuth app to add the challenge field in the authorization request and validate it in the response.
```

Comment:
Probably easier to use the Okta link rather than the standard here

```suggestion
The Okta configuration has no extra steps to support the [OAuth Authorization Code flow with Proof Key for Code Exchange](/docs/concepts/oauth-openid/#authorization-code-flow-with-pkce-flow) (PKCE). Okta handles PKCE automatically if the OAuth request includes the challenge. However, you must properly configure your external OAuth app to add the challenge field in the authorization request and validate it in the response.
```

Reply:
in next commit